### PR TITLE
vim 8.1.0001

### PR DESCRIPTION
**Pull request by martinwalsh**
* bumps version of vim to 8.1.0000
* adds an alias for vim@8.1
* vim@8.0: renamed from vim

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

First time contributor here ... this version bump would seem slightly more complicated than the normal kind, even considering my n00bness, so I beg your forgiveness for any oversights or misdeeds. Thanks in advance for your assistance!

P.S. The `brew bump-formula-pr` method failed for me with `Error: Unable to fork: !`. I suspect it's something to do with my `hub` config, but I haven't dug in. 

P.P.S. I did notice the admonition to only update vim for patch levels that are multiples of 50, which I assumed would include this one. But I do recognize this is perhaps too new to include now, in which case please feel free to close this PR with impunity (or for any reason, for that matter!).

### CODE REVIEW

File: formula_renames.json [hunk 1]
```diff
@@ -190,6 +190,7 @@
   "v8-315": "v8@3.15",
   "varnish4": "varnish@4",
   "vim74": "vim@7.4",
+  "vim80": "vim@8.0",
```
Comment (commitay) [inline formula_renames.json]: Remove this.

**Review by commitay — DISMISSED** [review]
Please remove the `vim@8.0` formula

File: formula_renames.json (hunk 1 above)
Comment (martinwalsh) [reply formula_renames.json]: Thanks @commitay

**Comment by martinwalsh** [comment]
> Please remove the vim@8.0 formula

I should remove `Formula/vim@8.0.rb` also? sorry for the confusion.

**Comment by commitay** [comment]
Yes.

**Comment by martinwalsh** [comment]
@commitay How about the alias (symlink)? `Aliases/vim@8.0 → Aliases/vim@8.1` is that copacetic?

Thanks a bunch for the guidance.

**Comment by commitay** [comment]
The alias rename is correct.

**Comment by yitsushi** [comment]
Can you update with the `0001` patched version?

https://github.com/vim/vim/releases/tag/v8.1.0001

```
Problem:    The netrw plugin does not work.
Solution:   Make it accept version 8.x.
```

**Comment by fxcoudert** [comment]
Thanks @martinwalsh for the pull request!

**Pull request closed without merging**